app/directives/proposal/proposal.directive.js

```diff
@@ -187,9 +184,15 @@ angular.module('dashboard')
                     setMode(PROP.MODE.OPEN);
                 }
                 else if (action === PROP.BTN.CANCEL.action) {
+                    if ($scope.proposal.isNew) {
```

Comment:
BTW something for the future: this if-elseif structure looks like it should be a switch-case